lib/rules/no-top-level-side-effects.ts

```diff
@@ -290,6 +291,29 @@ export const noTopLevelSideEffects: Rule.RuleModule = {
         }
       },
 
+      ObjectExpression: (node) => {
+        if (isTopLevel(node)) {
+          node.properties.forEach(property => {
+            if (property.type === 'SpreadElement') {
+              checkExpression(context, options, node, property.argument);
+              return;
+            }
```

Comment:
I'm inclined to say the spread operator should be considered a side effect because it **merges** two objects.

Reply:
hmm yea it probably is considered fine if `allowDerived` is `true` 

```ts
// ✅ ok if allowDerived === true
const bar = { a: 'b' };
const foo = { ...bar, c: 'd' };

// ❌ not ok because it invokes a function call - unless the function is allowlisted:
const foo = { ...getFoo() };
```